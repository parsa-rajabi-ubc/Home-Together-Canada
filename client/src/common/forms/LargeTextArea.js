/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.18
 *
 * @Description: Large text area form Component. Returns a text area box for larger string input.
 *
 */
import React from 'react';
import PropTypes from "prop-types";
import Asterisk from "./Asterisk";
import Tooltip from "./Tooltip";

function LargeTextArea(props) {
    const {label, name, placeholder, required, onChange, toolTipID, toolTipText} = props;
    return (
        <div>
            <label className={"label"}> {label} </label>{(required ? <Asterisk/> : '')}
            {toolTipID && <Tooltip text={toolTipText} toolTipID={toolTipID}/>}
            <textarea className={"input "} name={name} rows="5" cols="50" placeholder={placeholder}
                      onChange={onChange}/>
        </div>
    );
}

LargeTextArea.propTypes = {
    label: PropTypes.string.isRequired,
    toolTipText: PropTypes.string,
    toolTipID: PropTypes.string,
    name: PropTypes.string,
    required: PropTypes.bool,
    placeholder: PropTypes.string,
    onChange: PropTypes.func
};

export default LargeTextArea;