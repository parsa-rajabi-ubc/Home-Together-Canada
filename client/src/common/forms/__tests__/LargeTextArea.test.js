/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.19
 *
 * @Description: Phone number input fields Component Snapshot test
 *
 */
import React from 'react';
import renderer from  'react-test-renderer'
import LargeTextArea from "../LargeTextArea";

describe('LargeTextArea', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties", () => {
            // given
            const label = 'testLabelString1';
            const name = 'testLabelString2';
            const placeholder = 'testLabelString3';
            const onChangeMock = jest.fn();

            //when
            const component = renderer.create(<LargeTextArea label={label} name={name} placeholder={placeholder} onChange={onChangeMock} />).toJSON();

            //then
            expect(component).toMatchSnapshot();
        });
    })
})