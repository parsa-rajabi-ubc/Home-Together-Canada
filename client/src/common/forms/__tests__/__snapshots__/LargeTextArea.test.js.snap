// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeTextArea Snapshot test should render correctly regardless of properties 1`] = `
<div>
  <label
    className="label"
  >
    testLabelString1
  </label>
  
  <textarea
    className="undefined input"
    cols="50"
    name="testLabelString2"
    onChange={[MockFunction]}
    placeholder="testLabelString3"
  />
</div>
`;
